src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs

```diff
@@ -316,14 +309,16 @@ private struct TagEnumerationState : IActivityEnumerator<KeyValuePair<string, ob
 
             public bool ForEach(KeyValuePair<string, object> activityTag)
             {
-                if (activityTag.Value is Array)
-                {
-                    ProcessJaegerTagArray(ref this.Tags, activityTag);
-                }
-                else if (activityTag.Value != null)
+                if (activityTag.Value != null)
                 {
                     var key = activityTag.Key;
                     var jaegerTag = activityTag.ToJaegerTag();
+
+                    if (jaegerTag.Equals(default(JaegerTag)))
```

Comment:
Does that actually work? If it is calling the `Equals(object)` version, it will always be false (I think). Basically box the two values and then do a reference check. I don't think `JaegerTag` implements `IEquatable<JaegerTag>`. Could implement that, or make it a `record struct` to auto-generate it, BUT, all of those things are kind of slow. Could we make the API more like...

```csharp
if (!activityTag.TryToJaegerTag(out JaegerTag jaegerTag))
{
   return;
}
```

That would be faster and more expressive of what we're trying to accomplish I think.

Reply:
Yea I had it on my mind to convert all these to `TryBlahBlah` methods. At first I thought I might be able to take that on separately, but agree this code probably doesn't work, so I guess makes sense to `TryBlahBlah` now.

Reply:
Done. 💥 or should I say `Try💥`